src/argilla/server/contexts/datasets.py

```diff
@@ -591,55 +589,61 @@ async def _validate_vector(
     return vectors_settings
 
 
+async def _create_record(
+    db: "AsyncSession", dataset: Dataset, record_create: RecordCreate, caches: Dict[str, Any]
+) -> Tuple[Record, List[Vector]]:
+    _validate_record_fields(dataset, fields=record_create.fields)
+    caches["metadata_properties_cache"] = await _validate_record_metadata(
+        db, dataset, record_create.metadata, caches["metadata_properties_cache"]
+    )
+    record_responses, caches["users_ids_cache"] = await _build_record_responses(
+        db, dataset, record_create, caches["users_ids_cache"]
+    )
+    record_suggestions, caches["questions_cache"] = await _build_record_suggestions(
+        db, record_create, caches["questions_cache"]
+    )
+    record_vectors, caches["vectors_settings_cache"] = await _build_record_vectors(
+        db,
+        dataset,
+        record_create,
+        build_vector_func=lambda value, vector_settings_id: Vector(value=value, vector_settings_id=vector_settings_id),
+        cache=caches["vectors_settings_cache"],
+    )
+    record = Record(
+        fields=record_create.fields,
+        metadata_=record_create.metadata,
+        external_id=record_create.external_id,
+        dataset_id=dataset.id,
+        responses=record_responses,
+        suggestions=record_suggestions,
+        vectors=record_vectors,
+    )
+    return record, record_vectors
```

Comment:
It is really necessary to return `record_vectors`? You are already returning the record with them inside.

Reply:
yes, totally unnecessary